vms/rpcchainvm/vm_test.go

```diff
@@ -255,3 +256,39 @@ func TestNewHTTPHandler(t *testing.T) {
 	handler.ServeHTTP(w, httptest.NewRequest(http.MethodGet, "/", nil))
 	require.Equal(http.StatusOK, w.Code)
 }
+
+// TestConvertNetworkUpgrades_AllFieldsHandled ensures that all fields in
+// upgrade.Config are properly handled when converting from the gRPC protobuf message.
+func TestConvertNetworkUpgrades_AllFieldsHandled(t *testing.T) {
+	wantConfig := upgrade.Default
+
+	// convert to protobuf
+	pbUpgrades := &vmpb.NetworkUpgrades{
+		ApricotPhase_1Time:            grpcutils.TimestampFromTime(wantConfig.ApricotPhase1Time),
+		ApricotPhase_2Time:            grpcutils.TimestampFromTime(wantConfig.ApricotPhase2Time),
+		ApricotPhase_3Time:            grpcutils.TimestampFromTime(wantConfig.ApricotPhase3Time),
+		ApricotPhase_4Time:            grpcutils.TimestampFromTime(wantConfig.ApricotPhase4Time),
+		ApricotPhase_4MinPChainHeight: wantConfig.ApricotPhase4MinPChainHeight, // not passed through the RPC protocol
+		ApricotPhase_5Time:            grpcutils.TimestampFromTime(wantConfig.ApricotPhase5Time),
+		ApricotPhasePre_6Time:         grpcutils.TimestampFromTime(wantConfig.ApricotPhasePre6Time),
+		ApricotPhase_6Time:            grpcutils.TimestampFromTime(wantConfig.ApricotPhase6Time),
+		ApricotPhasePost_6Time:        grpcutils.TimestampFromTime(wantConfig.ApricotPhasePost6Time),
+		BanffTime:                     grpcutils.TimestampFromTime(wantConfig.BanffTime),
+		CortinaTime:                   grpcutils.TimestampFromTime(wantConfig.CortinaTime),
+		CortinaXChainStopVertexId:     wantConfig.CortinaXChainStopVertexID[:], // not passed through the RPC protocol
+		DurangoTime:                   grpcutils.TimestampFromTime(wantConfig.DurangoTime),
+		EtnaTime:                      grpcutils.TimestampFromTime(wantConfig.EtnaTime),
+		FortunaTime:                   grpcutils.TimestampFromTime(wantConfig.FortunaTime),
+		GraniteTime:                   grpcutils.TimestampFromTime(wantConfig.GraniteTime),
+		HeliconTime:                   grpcutils.TimestampFromTime(wantConfig.HeliconTime),
+	}
```

Comment:
This is copied from `vm_client.go`. This should be made into a helper (similarly to how `convertNetworkUpgrades` is a helper the other way).

Reply:
Turned into a helper!